internal/db/db.go

```diff
@@ -4,13 +4,78 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
+	"fmt"
 	"time"
 
 	"github.com/google/uuid"
 	"github.com/jackc/pgx/v4"
 	"github.com/jackc/pgx/v4/pgxpool"
 )
 
+// A time interval represent an amount of time unit to be subtracted or added to a date.
+type TimeInterval struct {
+	Operator TimeOperator // Plus or Minus
+	Amount   int          // The amount of the following unit
+	Unit     TimeUnit     // Unit of time
+}
+
+type TimeUnit int     // represents a Unit of time, such as second or hour.
+type TimeOperator int // represents the Plus or Minus operator
+
+var TimeIntervalInvalidTimeUnit = errors.New("Invalid time unit for the time interval")
+var TimeIntervalInvalidOperator = errors.New("Invalid operator for the time interval")
+var TimeIntervalInvalidAmount = errors.New("The count must be greater than 0 for the time interval")
+
+const (
+	Second TimeUnit = iota
+	Minute TimeUnit = iota
+	Hour   TimeUnit = iota
+	Day    TimeUnit = iota
+	Week   TimeUnit = iota
+	Month  TimeUnit = iota
+	Year   TimeUnit = iota
+)
+
+const (
+	Plus  TimeOperator = iota
+	Minus TimeOperator = iota
+)
+
+func (tI *TimeInterval) ToString() (string, error) {
+	var sUnit = ""
+	switch tI.Unit {
+	case Second:
+		sUnit = "Second"
+	case Minute:
+		sUnit = "Minute"
+	case Hour:
+		sUnit = "Hour"
+	case Day:
+		sUnit = "Day"
+	case Week:
+		sUnit = "Week"
+	case Month:
+		sUnit = "Month"
+	case Year:
+		sUnit = "Year"
+	default:
+		return "", TimeIntervalInvalidTimeUnit
+	}
+	var sign = ""
+	switch tI.Operator {
+	case Plus:
+		sign = "+"
+	case Minus:
+		sign = "-"
+	default:
+		return "", TimeIntervalInvalidOperator
+	}
+	if tI.Amount <= 0 {
+		return "", TimeIntervalInvalidAmount
+	}
+	return fmt.Sprintf("%s interval '%d %s'", sign, tI.Amount, sUnit), nil
+}
+
```

Comment:
I think this is a bit too much for what we need, and furthermore pgx4 supports time.Duration to interval https://github.com/jackc/pgtype/blob/master/interval.go, so if we want something with a bit more functionality let's use `time.Duration`.